code/presets/react-webpack/src/loaders/react-docgen-loader.ts

```diff
@@ -11,6 +11,11 @@ import type { LoaderContext } from 'webpack';
 import type { Handler, NodePath, babelTypes as t, Documentation } from 'react-docgen';
 import { logger } from '@storybook/node-logger';
 
+import {
+  ReactDocgenResolveError,
+  defaultLookupModule,
+} from '../../../../frameworks/react-vite/src/plugins/docgen-resolver';
```

Comment:
Is there anything we can do to clean things up? cc @valentinpalkovic

Reply:
I did this on purpose.

The only 2 places in our codebase that currently depend on `react-docgen` are:
![Screenshot 2024-02-29 at 17 49 37](https://github.com/storybookjs/storybook/assets/3070389/898b18dc-02e1-4bca-9362-1c0d2813a4e3)

So in order to extract this somewhere else, I'd need to introduce this to some library.
I understand this might look a bit weird, but it's bundled correctly.

I'm open to suggestions that do not create a new package.

Alternatively (I'm not a fan of this idea either) we could add this code to docs-tools;
but then docs tools gains a dependency to `react-docgen`, and with it, this:
![Screenshot 2024-02-29 at 17 53 04](https://github.com/storybookjs/storybook/assets/3070389/58e64926-3501-4f39-a7ae-5ebcd07f8a12)

If you feel like this is so bad we're better off duplicating the code, we could do that?

Reply:
Although this is kind of ugly (requiring files from the `react-vite` framework into the `react-webpack` preset things will compile.

I would rather vote to copy the code from `frameworks/react-vite/src/plugins/docgen-resolver` into a directory that is correctly accessible by both, the framework `react-vite` and the preset `react-webpack`, like in e.g. `@storybook/core-common` 

**or** 

we just duplicate the code in `../../../../frameworks/react-vite/src/plugins/docgen-resolver` which I think is okay if we don't want to move it into a sharable core package.

The third option is that things stay as is, but then `resolve` should be a direct dependency of the `react-webpack` package.

Reply:
@ndelangen Just to clarify. Are you talking about extracting `code/frameworks/react-vite/src/plugins/docgen-resolver.ts` into a core package? Because this file doesn't have any `react-docgen` dependencies or am i overlooking something?

Reply:
Can you define "correctly accessible by both"? @valentinpalkovic 
Placing the code in a shared package, like I mentioned that that would cause that package to gain dependencies it currently does not have.
You suggest `core-common`, I suggested `docs-tools`.

> Are you talking about extracting `code/frameworks/react-vite/src/plugins/docgen-resolver.ts` into a core package?
Yes we currently do not have a suitable package, hence I decided to just do this which minimizes duplicated code, and optimizes for easy of replaceability because I assume that at some point in the future this code will change upstream and updating it in 2 places going to be fragile, right?

But if you both agree that duplicating the code is better than 1 slightly ugly import, I'll change it.

Reply:
> `resolve` should be a direct dependency of the `react-webpack` package.

Ah, yes that's right. I missed that. 🙏

Reply:
I duplicated the code.